### PR TITLE
fix(mtp): correct backbone and sampler time measurement

### PR DESCRIPTION
**Pull request by deepsweet**
Related #1330, #1311, #1097.

A lot of people including me got confused with huge MTP sample timings. The actual problem is that due to mx.core lazy nature we've leaked massive logits computation to the sampler part until `mx.eval(verify_tok, bonus_tok)` or whatever first `.tolist()` happens.

This PR materializes it earlier so that it counts as `backbone_ms` properly:

```
INFO - MTP[0] finish=stop tokens=1886 cycles=973 accept=911/973 (93.6%) emits[init=2,draft=911,bonus=911,verify=62] timing[backbone=108907.4ms mtp=3935.5ms sample=401.0ms cache=54.1ms]
```

### CODE REVIEW

**Comment by jundot** [comment]
Thanks, this matches what @sangemaru traced in #1311. The single `mx.eval` barrier was materializing the whole verify forward into `sample_ms`, which is what confused everyone across #1097/#1311/#1330. Forcing `mx.eval(logits)` after the backbone attributes the cost correctly with negligible overhead. Merging for the next release. I'll refresh the now-stale split comment at 743-744 in a follow-up.
